src/component.rs

```diff
@@ -84,16 +75,24 @@ pub(crate) fn component_updatedir(sysroot: &str, component: &dyn Component) -> P
         .join(component.name())
 }
 
+/// Returns the name of the JSON file containing a component's available update metadata installed
+/// into the booted operating system root.
+fn component_update_data_name(component: &dyn Component) -> PathBuf {
+    Path::new(&format!("{}.json", component.name())).into()
```

Comment:
For future reference, direct `PathBuf::from<String>()` [exists](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#impl-From%3CString%3E).